test/model/game_state_test.dart

```diff
@@ -115,5 +117,48 @@ void main() {
     });
   });
 
+  group('Special cards', ()
+  {
+    test('should put Mutiny in next if too many resources', () {
+      const plunderAWreck = PlunderAWreck();
+      const aRivalShip = ARivalShip();
+      const mutiny = Mutiny();
+      final GameState state = makeGameState(
+          deck: [plunderAWreck, aRivalShip],
+          playerHand: [Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON,
+          Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON, Resource.DOUBLOON]
+      );
+
+      expect(state.nextCard(), mutiny);
```

Comment:
This test fails because the instances of `Mutiny` aren't identical - you're comparing one you constructed in this test to a different one which has been constructed inside the `nextCard` logic. In Dart and most OO languages, the default implementation for `equals` asserts that the two objects are actually the same identical instance in memory.

So we've got a few choices - if we overrode `equals` on the Mutiny card with our own implementation that'd do it, but since we only need this for tests I'd prefer changing the `expect` to just do a type check:

```
expect(state.nextCard(), new TypeMatcher<Mutiny>());
```

Reply:
Ah ok, that makes some sense, done.